### PR TITLE
Jonmv/token tell handler

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs or @tokle please review.

### CODE REVIEW

File: jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.yahoo.jdisc.http.filter.security.cloud;
+
+import com.yahoo.container.jdisc.HttpRequest;
+import com.yahoo.container.jdisc.HttpResponse;
+import com.yahoo.container.jdisc.ThreadedHttpRequestHandler;
+import com.yahoo.jdisc.http.filter.security.cloud.config.CloudTokenDataPlaneFilterConfig;
+import com.yahoo.restapi.SlimeJsonResponse;
+
+import javax.inject.Inject;
```
Comment (bjorncs) [inline jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java:9]: Wrong import!

File: jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.yahoo.jdisc.http.filter.security.cloud;
```
Comment (bjorncs) [inline jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java:1]: Missing copyright ;)

File: jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package com.yahoo.jdisc.http.filter.security.cloud;
```
Comment (bjorncs) [inline jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandlerTest.java:1]: Copyright

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java [hunk 4]
```diff
@@ -658,13 +660,24 @@ private void addCloudTokenSupport(DeployState state, ApplicationContainerCluster
...
         // Set as default filter for token port
         cluster.getHttp().getHttpServer().orElseThrow().getConnectorFactories().stream()
                 .filter(c -> c.getListenPort() == tokenPort).findAny().orElseThrow()
                 .setDefaultRequestFilterChain(tokenChain.getComponentId());
+
+        // Set up handler that tells what fingerprints are known to the container
+        class CloudTokenDataPlaneHandler extends Handler implements CloudTokenDataPlaneFilterConfig.Producer {
+            CloudTokenDataPlaneHandler() {
+                super(new ComponentModel("com.yahoo.jdisc.http.filter.security.cloud.CloudTokenDataPlaneHandler", null, "jdisc-security-filters", null));
+                addServerBindings(SystemBindingPattern.fromHttpPortAndPath(Defaults.getDefaults().vespaWebServicePort(), "cloud-token-data-plane-fingerprints"));
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java:676]: Suggestion: `cloud-token-data-plane-fingerprints` => `data-plane-tokens/v1`

**Review by bjorncs — CHANGES_REQUESTED** [review]

File: jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java (hunk 1 above)
Comment (jonmv) [reply jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java:9]: Thanks!

File: jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java (hunk 2 above)
Comment (jonmv) [reply jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandler.java:1]: OK ok ok .

File: jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandlerTest.java (hunk 3 above)
Comment (jonmv) [reply jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filter/security/cloud/CloudTokenDataPlaneHandlerTest.java:1]: :D

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java (hunk 4 above)
Comment (jonmv) [reply config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java:676]: Sure.

**Review by bjorncs — DISMISSED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java [hunk 5]
```diff
@@ -659,13 +660,24 @@ private void addCloudTokenSupport(DeployState state, ApplicationContainerCluster
...
         // Set as default filter for token port
         cluster.getHttp().getHttpServer().orElseThrow().getConnectorFactories().stream()
                 .filter(c -> c.getListenPort() == tokenPort).findAny().orElseThrow()
                 .setDefaultRequestFilterChain(tokenChain.getComponentId());
+
+        // Set up handler that tells what fingerprints are known to the container
+        class CloudTokenDataPlaneHandler extends Handler implements CloudTokenDataPlaneFilterConfig.Producer {
+            CloudTokenDataPlaneHandler() {
+                super(new ComponentModel("com.yahoo.jdisc.http.filter.security.cloud.CloudTokenDataPlaneHandler", null, "jdisc-security-filters", null));
+                addServerBindings(SystemBindingPattern.fromHttpPortAndPath(Defaults.getDefaults().vespaWebServicePort(), "data-plane-tokens/v1"));
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java:676]:
```suggestion
                addServerBindings(SystemBindingPattern.fromHttpPortAndPath(Defaults.getDefaults().vespaWebServicePort(), "/data-plane-tokens/v1"));
```

**Review by bjorncs — APPROVED** [review]

**Pull request merged**